### PR TITLE
Feature/endpoint method as api module

### LINKED ISSUE #44: Aito Configure to show first characters instead of last
When doing `aito configure` after already something has been configured, the command line displays a few characters of the configured instance. As all URLs tend to end the same way, it would be more beneficial to display first characters rather than the last.

image

### PR DESCRIPTION
**Pull request by tringm**
- Refactor Request classes, add Schema, Data, and Job API Request and Response Classes
- Return the appropriate request class in job_request
- Add use_job option in endpoint API methods
- Improve configure: prompt confirmation when profile exists, improved masking the instance url
- Refactor client, request, and response module into client subpackage

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -345,7 +346,7 @@ commonYAMLStructure:
       name: build and install package
       command: |
         . venv/bin/activate
-        ./scripts/deploy test.pypi --no-publish --bump-version dev
+        ./scripts/deploy test.pypi --no-publish --bump-version dev --no-check-git-tree
```
Comment (laaksonm) [inline .circleci/config.yml:349]: Are the `--no-check-git-tree` meant to be here?

**Review by laaksonm — COMMENTED** [review]
One question. :nerd_face:

File: .circleci/config.yml (hunk 1 above)
Comment (tringm) [reply .circleci/config.yml:349]: yes. Due to the JIRA. JIRA supposes to delete it "config" file but never does and the deploy command fails.

Comment (laaksonm) [reply .circleci/config.yml:349]: Check. :+1:

**Review by laaksonm — APPROVED** [review]
Ship it! :ship:

**Pull request merged**